### PR TITLE
add LineageOS plugin

### PR DESCRIPTION
**Pull request by Vince1171**
**DO NOT merge until   progressive-downloader is updated to 1.0.7**
### Current limitations:
 * No LineageOS Recovery support (I haven't found {yet} an API for it)
 * only download the latest available image

### Usage: 
Select the channel (always nightly)
```
options:
      - var: "channel"
        name: "LineageOS version"
        tooltip: "Which LineageOS channel do you want to install?"
        type: "select"
        remote_values:
          lineage_os:channels:
```
Install the latest LineageOS zip (need `adb` and a ORS compatible recovery)
```
steps:
      - actions:
          - lineage_os:install:
```
Download the latest zip with the name `lineageos_rootfs_DEVICE.zip`
(used automatically by `lineage_os:install`)
```
steps:
      - actions:
          - lineage_os:download:
```

### CODE REVIEW

File: src/core/plugins/lineage_os/api.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+const axios = require("axios");
+
+/** @module lineage_os */
+
+const baseURL = "https://download.lineageos.org/api/v1/";
+const deviceBuildTypeURL = `${baseURL}types/`;
+const rootfsDefaultName = "lineageos_rootfs_";
+
+
```
Comment (NeoTheThird) [inline src/core/plugins/lineage_os/api.js:28]:
```suggestion
```

**Review by NeoTheThird — CHANGES_REQUESTED** [review]
awesome!

- [ ] (currently blocked by #1552) update progressive-downloader to 0.1.7
- [ ] document new steps as was done in https://github.com/ubports/installer-configs/pull/70

**Comment by Vince1171** [comment]
As  #1552 is still in progress, can this be merged? (with ubports/installer-configs#85)

**Comment by Vince1171** [comment]
ping @NeoTheThird

**Comment by NeoTheThird** [comment]
My deepest apologies for the huge delay. Yes, let's go!

**Pull request merged**